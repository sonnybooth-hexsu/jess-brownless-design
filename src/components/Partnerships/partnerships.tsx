import React from "react"

type PartnershipProps = {
  name: string
  img: string
}

type PartnershipsProps = {
  images: Array<PartnershipProps>
}

export const Partnerships = ({ images }: PartnershipsProps) => (
  <div className="flex items-center">
    <p className="flex-shrink-0">Cosmetic brand partnerships:</p>
    {images?.map(({ name, img }, id) => (
      <div key={id} className="flex items-center justify-end w-full h-16 ml-2">
        {" "}
        <img
          data-testid="partnershipImg"
          className="h-full"
          src={img}
          alt={name}
        />
      </div>
    ))}
  </div>
)

export default Partnerships
