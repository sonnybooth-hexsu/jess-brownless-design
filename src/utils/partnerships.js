import placeholder from "../assets/images/placeholder-image.jpg"

export const Partnerships = [
  {
    name: "Placeholder 1",
    img: placeholder,
  },
  {
    name: "Placeholder 2",
    img: placeholder,
  },
  {
    name: "Placeholder 3",
    img: placeholder,
  },
  {
    name: "Placeholder 4",
    img: placeholder,
  },
  {
    name: "Placeholder 5",
    img: placeholder,
  },
]

export default Partnerships
