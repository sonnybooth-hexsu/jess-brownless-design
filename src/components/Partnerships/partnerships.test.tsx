/* eslint no-undef: 0 */

import React from "react"
import { render, cleanup } from "@testing-library/react"
import Partnerships from "./partnerships"
import partnerships from "../../utils/partnerships"

afterEach(() => {
  cleanup()
})

describe("<Partnerships />", () => {
  it("should match snapshot", () => {
    const { container } = render(<Partnerships images={partnerships} />)

    expect(container).toMatchSnapshot()
  })

  it("has partnership image", () => {
    const { queryAllByTestId } = render(<Partnerships images={partnerships} />)

    const images = queryAllByTestId("partnershipImg")

    expect(images).toHaveLength(partnerships.length)
  })
})
