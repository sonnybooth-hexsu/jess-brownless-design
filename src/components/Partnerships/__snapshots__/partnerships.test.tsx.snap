// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Partnerships /> should match snapshot 1`] = `
<div>
  <div
    class="flex items-center"
  >
    <p
      class="flex-shrink-0"
    >
      Cosmetic brand partnerships:
    </p>
    <div
      class="flex items-center justify-end w-full h-16 ml-2"
    >
       
      <img
        alt="Placeholder 1"
        class="h-full"
        data-testid="partnershipImg"
        src="test-file-stub"
      />
    </div>
    <div
      class="flex items-center justify-end w-full h-16 ml-2"
    >
       
      <img
        alt="Placeholder 2"
        class="h-full"
        data-testid="partnershipImg"
        src="test-file-stub"
      />
    </div>
    <div
      class="flex items-center justify-end w-full h-16 ml-2"
    >
       
      <img
        alt="Placeholder 3"
        class="h-full"
        data-testid="partnershipImg"
        src="test-file-stub"
      />
    </div>
    <div
      class="flex items-center justify-end w-full h-16 ml-2"
    >
       
      <img
        alt="Placeholder 4"
        class="h-full"
        data-testid="partnershipImg"
        src="test-file-stub"
      />
    </div>
    <div
      class="flex items-center justify-end w-full h-16 ml-2"
    >
       
      <img
        alt="Placeholder 5"
        class="h-full"
        data-testid="partnershipImg"
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
